### PR TITLE
sidechain createGroup update

### PR DESCRIPTION
**Pull request by wilbarnes**
- moved interfaces each into own file (were previously in Group.sol)
- added npm package for 'sidechain/'
- reverted package.json in root directory back to previous values (i changed solc & openzeppelin versions)
- createGroup contract done and needs to be reviewed
- docstrings so that contract is easier to understand and reader can understand my thinking
- small test suite (only 2 right now) that i will continue to expand

### CODE REVIEW

File: sidechain/contracts/group/Group.sol [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    function addMember(
+        uint256 _groupId, 
+        address _sender, 
+        uint8 _role
+    )
+        internal
+        returns (bool)
+    {
+        storageContract.putUintValue(keccak256(
```
[the remark this answers is not in the document]
Comment (wilbarnes) [reply sidechain/contracts/group/Group.sol]: fixed, added _sender

File: sidechain/test/test_group.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        let sig             = await sign(testprivkey, hash);
+        
+        let newGroup        = await instance.createGroup(metadataLocator, sig, incorrectNonce);
+        let newSequence     = await instance.sequence.call();
+
+        assert.equal(newSequence, 1);
+    })
+});
+
+async function sign(pk, message) {
```
[the remark this answers is not in the document]
Comment (wilbarnes) [reply sidechain/test/test_group.js]: i updated the tests to use web3.eth.sign. is that all that needs to be done to ensure it works in travis? no need to use HDWalletProvider?

**Comment by wilbarnes** [comment]
testing suite is in place. i also just adapted Craig's storage test file, it just needed a few web3js version changes. started writing tests for the invitation logic, and i'll review with greg in the morning. feeling confident i can get the invitation work drafted up by the end of the week

File: sidechain/migrations/2_deploy_dependency_contracts.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+let Group = artifacts.require('Group');
+let Storage = artifacts.require('Storage');
+let OnlyOwnerAdminController = artifacts.require('OnlyOwnerAdminController');
+
+async function performMigration(deployer, network, accounts) {
+    await deployer.deploy(Storage);
```
Comment (joshorig) [inline sidechain/migrations/2_deploy_dependency_contracts.js]: @wilbarnes 
Just realised currently this file is deploying all the contracts.
In this case I believe `Group` wouldn't be a dependency but a main contract.

So similar to mainchain folder I would suggest deploy the `Storage` and `OnlyOwnerAdminController` files here and then create a third deployment file `3_deploy_logic_contracts.js` Which then deploys the `Group`contract, this way deployment can upgrade the Group contract and then use the same storage.

Just double check with @craigwilliams84  on the above incase I got that wrong

File: sidechain/test/test_group.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshorig) [reply sidechain/test/test_group.js]: Should be ok yes, this is how we have it on mainchain

File: sidechain/migrations/2_deploy_dependency_contracts.js (hunk 3 above)
Comment (craigwilliams84) [reply sidechain/migrations/2_deploy_dependency_contracts.js]: Yep, thats right.  If the `Group` contract is deployed in the same migration script as `Storage` then it won't be possible to deploy `Group` without deploying a new `Storage` contract as well.  They should be separated for sure.

**Pull request merged**